features/netsocket/NetworkInterface.cpp

```diff
@@ -89,3 +89,132 @@ nsapi_error_t NetworkInterface::set_blocking(bool blocking)
     return NSAPI_ERROR_UNSUPPORTED;
 }
 
+/* The stack of functions providing support for finding network interfaces */
+
+#include "EthInterface.h"
+#include "WiFiInterface.h"
+#include "CellularBase.h"
+#include "MeshInterface.h"
```

Comment:
Having #include lines in the middle of a C file looks a bit weird, is this a leftover from rebasing?

Reply:
It's possibly a sign that the thing should be split into two files. Everything above this point is the "core" NetworkInterface stuff which knows nothing about derived types.

This point and below is the "dispatch to known derived types" logic. Which arguably is a piece of "system" code providing the built-in behaviour for the 4 core types.

Possibly move this half of the file to NetworkInterfaceDefaults.cpp?